backend/src/test/java/com/github/astridottmann/backend/controllers/CompareRoutesIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+                                "numberOfPersons": 1,
+                                "oneWay": false,
+                                "vehicle": {
+                                    "type": "car",
+                                    "co2Emission": 253.3,
+                                    "fuel": "petrol",
+                                    "carSize": "large"
+                                },
+                                "co2EmissionRoute": 268.93
+                            },
+                            {
+                                "id": "456",
+                                "start": "Hamburg",
+                                "destination": "Frankfurt",
+                                "distance": 492,
+                                "numberOfPersons": 1,
+                                "oneWay": false,
+                                "vehicle": {
+                                    "type": "publicTransport",
+                                    "co2Emission": 46.0,
+                                    "distanceLevel": "longDistance",
+                                    "meansOfTransport": "train"
+                                },
+                                "co2EmissionRoute": 45.26
+                            }
+                        ],
+                     "comparisonResults": {
+                            "resultRouteOne": 268.93,
+                            "resultRouteTwo": 45.26,
+                            "difference": 223.67
+                        }
+                    }
+                """;
+    }
+
+    @Test
+    void getAllCompareRoutes_shouldReturnAllCompareRoutesFromTheRepository() throws Exception {
+        compareRoutesRepository.save(testCompareRoutes);
+        List<CompareRoutes> expectedBody = new ArrayList<>(List.of(testCompareRoutes));
+        expectedBodyJson = objectMapper.writeValueAsString(expectedBody);
+
+        mockMvc.perform(get("/api/compare"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(expectedBodyJson));
+    }
+
+    @Test
+    void getAllCompareRoutes_shouldReturnEmptyList_whenRepositoryIsEmpty() throws Exception {
+        mockMvc.perform(get("/api/compare"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
+                        []
+                        """));
+    }
+
+    @Test
+    void addComparison() throws Exception {
+
+                mockMvc.perform(post("/api/compare")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(testComparedJson))
+                        .andExpect(status().isOk())
+                        .andExpect(content().json(testCompareRoutesWithoutIdJson))
+                        .andExpect(jsonPath("$.id").isNotEmpty())
+                        .andReturn()
+                        .getResponse()
+                        .getContentAsString();
+
```

Comment:
could be removed, because you don't assign and use the return value.

Reply:
You're right! Forgot it :-) 
Actually I also wanted to check, if the new object is in the repo (that's why it's there), but then I let it.